### PR TITLE
[TCD-2485] Clarify how to update RDC jobs

### PR DESCRIPTION
**Pull request by lysannep**
### Description
https://saucedev.atlassian.net/browse/TCD-2485

### Motivation and Context
Clarifies that the update_job API method does not work for RDC - the Selenium JS Executor should be used instead. Also adds keywords to the Test Annotation doc to enhance SEO.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [ ] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/main/docs/contributing.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/basics/test-config-annotation/test-annotation.md [hunk 1]
```diff
@@ -16,7 +20,7 @@ export const Highlight = ({children, color}) => ( <span style={{
...
+Test annotation refers to adding information to your tests after they have completed, such as setting a name, build number, tag, and Pass/Fail status. These annotations are useful for managing your tests and builds (for example, when searching and sorting tests in your [Archives](/test-results/archived-test-results)). You can add annotations with our [REST API](/basics/test-config-annotation/test-annotation/#sauce-labs-rest-api) or the [Selenium JavaScript Executor](/basics/test-config-annotation/test-annotation/#selenium-javascript-executor). You can also use [sample test frameworks](https://github.com/saucelabs-training) to automatically add annotations to your tests.
```
Comment (wswebcreation) [inline docs/basics/test-config-annotation/test-annotation.md:23]: > Test annotation refers to adding information to your tests after they have completed,....

This is only correct for the API, the JS executor can only be used **during** a test. Can we clarify this better?

File: docs/basics/test-config-annotation/test-annotation.md [hunk 2]
```diff
@@ -149,10 +153,12 @@ public class TextInputTest extends TestBase {
...
+You can manage your tests more effectively from your Dashboard and Archives with annotations. The Sauce Labs REST API includes an [update_job](/dev/api/jobs/#update-a-job) method that you can use to set a name, tags, pass/fail status, and custom data for your test after it runs. To automate test annotation with this method, you'll want to create a simple set of functions to perform the put request for you. We've developed a [Java library](https://github.com/saucelabs/saucerest-java) to do just that, with examples for [Python](https://gist.github.com/1644439) and [Ruby](https://gist.github.com/DylanLacey/5218959) on GitHub.
```
Comment (wswebcreation) [inline docs/basics/test-config-annotation/test-annotation.md:156]: This sentence doesn't seem right

>  We've developed a [Java library](https://github.com/saucelabs/saucerest-java) to do just that, with examples for [Python]....

You don't develop a lib for Java with example for Python or Ruby. We might want to say that we developed a lib for Java, and we **have** examples for Python and Ruby

File: docs/basics/test-config-annotation/test-annotation.md (hunk 1 above)
Comment (wswebcreation) [reply docs/basics/test-config-annotation/test-annotation.md:23]: Secondly we might want to mention that the update-job API can only be used for VDC, not for RDC

File: docs/basics/test-config-annotation/test-annotation.md [hunk 3]
```diff
@@ -149,10 +153,12 @@ public class TextInputTest extends TestBase {
...
 
-:::note
-Adding Pass/Fail Status and Build Numbers to Test Results with Frameworks<br/>
+:::note RDC Jobs
+The [update_job](/dev/api/jobs/#update-a-job) method is not supported for RDC. To update a job for real devices, use the [Selenium JavaScript Executor](/basics/test-config-annotation/test-annotation/#selenium-javascript-executor).
```
Comment (wswebcreation) [inline docs/basics/test-config-annotation/test-annotation.md:159]: See my remark about the update job API, maybe remove the above note and mention that it **can't** be used for RDC

File: docs/dev/api/jobs.md [hunk 4]
```diff
@@ -274,6 +274,10 @@ curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" --location \
 
 Edit job attributes based on parameters passed in the request, including setting the status of the job. Any parameter for which a new value is provided in the request will replace the existing value. For example, if you provide a set of tags, they will not be added to the current tags; they will replace them, so make sure you pass the entire set you wish to assign.
 
+:::note RDC Jobs
+This call is specific to jobs running in simulation. To update a job for real devices, use the [Selenium JavaScript Executor](/basics/test-config-annotation/test-annotation/#selenium-javascript-executor).
```
Comment (wswebcreation) [inline docs/dev/api/jobs.md:278]: It's not only the update job status, also the update job name. Can we also mention that here?

**Review by wswebcreation — CHANGES_REQUESTED** [review]
Hi @lysannep 

Thanks for the PR, see my comments for some feedback/changes

**Review by aoleszyn — APPROVED** [review]
LGTM

File: docs/basics/test-config-annotation/test-annotation.md (hunk 1 above)
Comment (lysannep) [reply docs/basics/test-config-annotation/test-annotation.md:23]: I added a "Virtual Devices Only" badge to clarify.

File: docs/basics/test-config-annotation/test-annotation.md (hunk 2 above)
Comment (lysannep) [reply docs/basics/test-config-annotation/test-annotation.md:156]: Updated - reformatted as bulleted list of examples in a details dropdown.

File: docs/basics/test-config-annotation/test-annotation.md (hunk 3 above)
Comment (lysannep) [reply docs/basics/test-config-annotation/test-annotation.md:159]: Removed.

**Review by wswebcreation — APPROVED** [review]
LGTM now, thanks @lysannep

**Comment by lysannep** [comment]
Thanks @wswebcreation!

**Pull request merged**